File: nagasena/src/org/openexi/proc/events/EXIEventDTD.java
package org.openexi.proc.events;

import org.openexi.proc.common.BinaryDataSource;
import org.openexi.proc.common.EventType;
import org.openexi.proc.common.EventDescription;
import org.openexi.schema.Characters;

public final class EXIEventDTD implements EventDescription {
  
  private final String m_name;
  private final String m_publicId;
  private final String m_systemId;
  private final Characters m_text;
  private final EventType m_eventType;

  public EXIEventDTD(String name, String publicId, String systemId, Characters text, EventType eventType) {
    assert eventType.itemType == EventType.ITEM_DTD;
    m_name = name;
    m_publicId = publicId;
    m_systemId = systemId;
    m_text = text;
    m_eventType = eventType;
  }

  ///////////////////////////////////////////////////////////////////////////
  // Accessors
  ///////////////////////////////////////////////////////////////////////////
  
  public String getPublicId() {
    return m_publicId;
  }
  
  public String getSystemId() {
    return m_systemId;
  }

  ///////////////////////////////////////////////////////////////////////////
  // Implementation of EXIEvent interface
  ///////////////////////////////////////////////////////////////////////////

  public byte getEventKind() {
    return EventDescription.EVENT_DTD;
  }
  
  public String getURI() {
    return null;
  }
  
  public String getName() {
    return m_name;
  }

  public int getURIId() {
    return -1;
  }

  public int getNameId() {
    return -1;
  }

  public String getPrefix() {
    return null;
  }

  public Characters getCharacters() {
    return m_text;
  }

  public BinaryDataSource getBinaryDataSource() {
    return null;
  }

  public final EventType getEventType() {
    return m_eventType;
  }

}

File: nagasenacs/src/Nagasena/Proc/Events/EXIEventDTD.cs
using System.Diagnostics;

using BinaryDataSource = Nagasena.Proc.Common.BinaryDataSource;
using EventType = Nagasena.Proc.Common.EventType;
using EventDescription = Nagasena.Proc.Common.EventDescription;
using EventDescription_Fields = Nagasena.Proc.Common.EventDescription_Fields;
using Characters = Nagasena.Schema.Characters;

namespace Nagasena.Proc.Events {

  /// <exclude/>
  public sealed class EXIEventDTD : EventDescription {

    private readonly string m_name;
    private readonly string m_publicId;
    private readonly string m_systemId;
    private readonly Characters m_text;
    private readonly EventType m_eventType;

    public EXIEventDTD(string name, string publicId, string systemId, Characters text, EventType eventType) {
      Debug.Assert(eventType.itemType == EventType.ITEM_DTD);
      m_name = name;
      m_publicId = publicId;
      m_systemId = systemId;
      m_text = text;
      m_eventType = eventType;
    }

    ///////////////////////////////////////////////////////////////////////////
    // Accessors
    ///////////////////////////////////////////////////////////////////////////

    public string PublicId {
      get {
        return m_publicId;
      }
    }

    public string SystemId {
      get {
        return m_systemId;
      }
    }

    ///////////////////////////////////////////////////////////////////////////
    // Implementation of EXIEvent interface
    ///////////////////////////////////////////////////////////////////////////

    public sbyte EventKind {
      get {
        return EventDescription_Fields.EVENT_DTD;
      }
    }

    public string URI {
      get {
        return null;
      }
    }

    public string Name {
      get {
        return m_name;
      }
    }

    public int URIId {
      get {
        return -1;
      }
    }

    public int NameId {
      get {
        return -1;
      }
    }

    public string Prefix {
      get {
        return null;
      }
    }

    public Characters Characters {
      get {
        return m_text;
      }
    }

    public BinaryDataSource BinaryDataSource {
      get {
        return null;
      }
    }

    public EventType getEventType() {
      return m_eventType;
    }

  }

}